content/automate/ManageIQ/Infrastructure/VM/Transform/StateMachines/VmImport.class/__methods__/submit_vm_import.rb

```diff
@@ -0,0 +1,40 @@
+#
+# Description: This method initiates VM import to given infra provider
+#
+
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Transform
+          module StateMachines
+            class SubmitVmImport
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                provider = @handle.vmdb(:ext_management_system, @handle.root['provider_id'])
```

Comment:
Well this code is mostly forwarding all arguments to the import_vm backend where all necessary validation should be performed. I am not entirely sure if duplicating it here is worth it...